library/L2_HAL/communication/bluetooth/serial_bluetooth.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <chrono>
+#include <initializer_list>
+
+namespace sjsu
+{
+namespace bluetooth
+{
+/// The interface should be implemented by bluetooth peripherals that utilizes
+/// serial communication such as SPI or UART for the transferring of data.
+class SerialBluetooth
+{
+ public:
+  /// @returns True if the device currently has a connection with a another
+  ///          device.
+  virtual bool GetState() const = 0;
+
+  /// Disconnect from the current connection.
+  virtual void Disconnect() const = 0;
+
+  virtual void SendCommand(const char * command,
+                           const char * parameter           = nullptr,
+                           std::chrono::nanoseconds timeout = 10ms) const = 0;
```

Comment:
We may want to brain storm on this. I'm not sure how useful this interface is at the moment compared to just a straight driver. We should meetup next week to discuss this.